assets/js/components/GlobalFilters.vue

```diff
@@ -206,29 +222,47 @@ export default Vue.extend({
       return false;
     },
 
+    removeFilter(e) {
+      const remove_el = e.target.dataset.filter;
+      this.filters[remove_el] = null;
+    },
+    // chronologicalFilters holds the active filters ordered chronologically
     removeLastFilter() {
-      const $this=this;
-      window.addEventListener("keyup", function(e){
-        if (e.keyCode == 27) {
-          if($this.$el.querySelector('.list-filters')){
-          let last_filter = $this.$el.querySelector('.list-filters .filter-item:last-child');
-          if (!last_filter) return;
-          this.filters[last_filter.getAttribute('filter')] = null
-          }
-        }
-      })
+      let removedFilter = this.chronologicalFilters[this.chronologicalFilters.length-1];
+      this.filters[removedFilter.type] = null;
     },
-
-    resetFilters(){
+    
+    resetFilters() {
       for (const filter in this.filters) {
         this.filters[filter] = null
       }
+      this.chronologicalFilters = [];
+    },
+    // it is executed each time this.filters changes
+    // it will keep track of all filters applied or removed, chronologically
+    // if a previous same type filter is removed, then it will be removed from the list
+    // but if it is changed after remove, the new one will be added as the most recent
+    handleFilter(type, val, old) {
+      if(old) {
+        this.removeOneFilter(type, val);
+        if(val) {
+          this.chronologicalFilters.push({type, val});
+        }
+      } else {
+        if(val) {
+          this.chronologicalFilters.push({type, val});
+        }
+      }
     },
-  },
 
-  handleFilter(type, val, old) {
-    console.log(type, val, old)
+    removeOneFilter(type, val) {
+      for(var i=0; i<this.chronologicalFilters.length; i++) {
```

Comment:
if you loop, at least break on first find.
but there's no need to loop: either make this a simple stack and just `.indexOf()`, or else use `array.find()`. it's new in ES6 and we have it polyfilled.